### DISCUSSION TITLE
Support for Mi Silent Mouse.

### DISCUSSION THREAD

**Discussion by xplodr** [noah-nuebling/mac-mouse-fix#238] [Questions] [answered]
Hello!

When running mac-mouse-fix, it detects two side buttons from the Mi Silent Mouse (button 4 and 5), but unfortunately that's all. The buttons don't work in any application, including mac-mouse-fix itself (the menu of adding new actions doesn't notice pressing these two buttons).
Perhaps I am setting something wrong. Is there any solution or is it possible in the future? Thanks in advance!

**Comment by noah-nuebling** [comment]
Hi @xplodr!

I've heard from many people with the same issue as you. 
This seems to be a common issue with Xiaomi mice. 
From what I've heard, you should be able to use the side buttons when you connect the mouse via USB instead of Bluetooth. 

From what I've gathered, this issue occurs because the mouse tells the OS that it doesn't have side buttons at all when it's connected via Bluetooth. And I think in order to fix this you'd have to completely rewrite and replace the Apple system driver with a custom one that knows about this quirk and ignores the wrongly reported number of buttons. This is not really feasible to implement in Mac Mouse Fix unfortunately. 
But I plan to look into this some more at some point.  
Maybe there's a better solution where you can tell the Apple Driver that the mouse has more buttons or something. But of course I can't guarantee that!

Hope that clears things up, let me know if you have any more questions!

**Comment by xplodr** [comment]
Thanks for the reply and advice! 

According to my observations, Mac Mouse Fix correctly detects that the mouse has buttons 4 and 5, and immediately writes about it assigning them to the events "back" and "forward", but that's all - the program does not register clicks on these buttons.

So far I managed to solve the problem with SteerMouse. This utility somehow managed to get this mouse working - the buttons were detected and started working even via bluetooth. But this utility has another problem - it automatically detects that the mouse has too many buttons (18, if I remember correctly), but it is buttons 4 and 5 that end up being functional. I hope this information helps somehow.

I hope for a future fix in Mac Mouse Fix! :)

**Reply by noah-nuebling** [reply]
Oh that's interesting. SteerMouse does a lot of really tricky low-level magic, so this might be hard to replicate. But good to know that there's a way. Thanks!

I plan to look into this more eventually.

**Reply by xplodr** [reply]
Noah, thank you for your attention to this issue!

**Comment by noah-nuebling** [comment, marked as answer]
Hi! Marking this as answered since I'm moving the topic to Issue #121. Please continue discussion of Mi Silent Mouse Issues there. Thanks!

**Comment by KirillEraser** [comment]
Guys, I wrote an app that makes side buttons work on Xiaomi Mi Silent Button. Should work plug&play - check it out

https://github.com/KirillEraser/xiaomi-mouse-macOS-fix
